File: frontend/front/src/app/model/user-request.ts
import { UserRole } from "./enums/user_role";

export class UserRequest{
    private login: string;
    private password: string;
    private fullName: string;
    private email: string;
    private role: UserRole;

    constructor(login: string, password: string, fullName: string, email: string, role: UserRole){
            this.login = login;
            this.password = password
            this.fullName = fullName;
            this.email = email;
            this.role = role;
    }
    getLogin():string{
        return this.login;
    }
    setLogin(login: string){
        this.login = login;
    }
    getLPassword():string{
        return this.password;
    }
    setLPassword(password: string){
        this.password = password;
    }
    getFullName(): string {
        return this.fullName;
    }
    setFullName(fullName: string): void {
        this.fullName = fullName;
    }
    getEmail(): string {
        return this.email;
    }
    setEmail(email: string): void {
        this.email = email;
    }
    getRole(): UserRole {
        return this.role;
    }
    setRole(role: UserRole): void {
        this.role = role;
    }

    toJSON() {
        return {
            login: this.login,
            password: this.password,
            full_name: this.fullName, 
            email: this.email,
            role: this.role
        };
    }
}

File: back/dto/user_request.py
from pydantic import BaseModel

from enums.role import UserRole


class UserRequest(BaseModel):
    login: str
    password: str
    full_name: str
    email: str
    role: UserRole
    # def __init__(self, login: str, password: str, full_name: str, email: str, role: UserRole):
    #     self._login = login
    #     self._password = password
    #     self._full_name = full_name
    #     self._email = email
    #     self._role = role
    #
    # def login(self) -> str:
    #     return self._login
    #
    # def password(self) -> str:
    #     return self._password
    #
    # def full_name(self) -> str:
    #     return self._full_name
    #
    # def email(self) -> str:
    #     return self._email
    #
    # def role(self) -> UserRole:
    #     return self._role


class UserUpdate(UserRequest):
    id: int


class UserResponse(UserRequest):
    id: int
    orders: list
